fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/InternalBiomeData.java

```diff
@@ -0,0 +1,184 @@
+		OVERWORLD_RIVER_MAP.put(parent, river);
+		if (river != null) {
+			OVERWORLD_INJECTED_BIOMES.add(river);
+		}
+	}
+
+	/**
+	 * Allows players to naturally spawn in this biome
+	 *
+	 * @param biome
+	 */
+	public static void addSpawnBiome(Biome biome) {
+		SPAWN_BIOMES.add(biome);
+	}
+
+	public static Map<OverworldClimate, List<BiomeEntry>> getOverworldBaseBiomes() {
+		return OVERWORLD_BASE_BIOMES;
+	}
+
+	public static List<Biome> getOverworldInjectedBiomes() {
+		return OVERWORLD_INJECTED_BIOMES;
+	}
+
+	public static Set<Biome> getSpawnBiomes() {
+		return SPAWN_BIOMES;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldHills() {
+		return OVERWORLD_HILLS_MAP;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldShores() {
+		return OVERWORLD_SHORE_MAP;
+	}
+
+	public static Map<Biome, WeightedBiomePicker> getOverworldEdges() {
+		return OVERWORLD_EDGE_MAP;
+	}
+
+	public static Map<Biome, Biome> getOverworldRivers() {
+		return OVERWORLD_RIVER_MAP;
+	}
+
+	public static EnumMap<OverworldClimate, Double> getOverworldModdedWeightTotals() {
+		return OVERWORLD_MODDED_WEIGHT_TOTALS;
+	}
+
+	public static Map<Biome, VariantTransformer> getOverworldVariantTransformers() {
+		return OVERWORLD_VARIANT_TRANSFORMERS;
+	}
+	
+	@SuppressWarnings("deprecation")
+	public static double getOverworldBiomeWeight(OverworldClimate climate, Biome biome) {
+		Object2DoubleMap<Biome> climateMap = OVERWORLD_BIOME_WEIGHT_TOTALS.get(climate);
+		return climateMap.containsKey(biome) ? climateMap.get(biome) : 0;
+	}
+	
+	static {
+		// Add Vanilla Weights
+		Object2DoubleMap<Biome> cool_weight_totals = OVERWORLD_BIOME_WEIGHT_TOTALS.computeIfAbsent(OverworldClimate.COOL, map -> new Object2DoubleOpenHashMap<>());
+		Object2DoubleMap<Biome> dry_weight_totals = OVERWORLD_BIOME_WEIGHT_TOTALS.computeIfAbsent(OverworldClimate.DRY, map -> new Object2DoubleOpenHashMap<>());
+		Object2DoubleMap<Biome> snowy_weight_totals = OVERWORLD_BIOME_WEIGHT_TOTALS.computeIfAbsent(OverworldClimate.SNOWY, map -> new Object2DoubleOpenHashMap<>());
+		Object2DoubleMap<Biome> temperate_weight_totals = OVERWORLD_BIOME_WEIGHT_TOTALS.computeIfAbsent(OverworldClimate.TEMPERATE, map -> new Object2DoubleOpenHashMap<>());
+		
+		cool_weight_totals.put(Biomes.FOREST, 1);
+		cool_weight_totals.put(Biomes.MOUNTAINS, 1);
+		cool_weight_totals.put(Biomes.PLAINS, 1);
+		cool_weight_totals.put(Biomes.TAIGA, 1);
+		
+		dry_weight_totals.put(Biomes.DESERT, 3);
+		dry_weight_totals.put(Biomes.SAVANNA, 2);
+		dry_weight_totals.put(Biomes.PLAINS, 1);
+		
+		snowy_weight_totals.put(Biomes.SNOWY_TUNDRA, 3);
+		snowy_weight_totals.put(Biomes.SNOWY_TAIGA, 1);
+		
+		temperate_weight_totals.put(Biomes.FOREST, 1);
+		temperate_weight_totals.put(Biomes.MOUNTAINS, 1);
+		temperate_weight_totals.put(Biomes.PLAINS, 1);
+		temperate_weight_totals.put(Biomes.DARK_FOREST, 1);
+		temperate_weight_totals.put(Biomes.BIRCH_FOREST, 1);
+		temperate_weight_totals.put(Biomes.SWAMP, 1);
```

Comment:
These weights should be computed by iterating the vanilla array. It is currently very fragile regarding minecraft updates and other edits.

Reply:
Prospector removed the whole thing lol

I did this for my mod TheBiomeOverhaul and other mods that overhaul biomes but I can just use a mixin instead I guess